chapter8.adoc

```diff
@@ -0,0 +1,30 @@
+[[chapter8]]
+[[Smsdedbg]]
+== `Smsdedbg`: Supervisor Domain External Debug
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "External Debug for Supervisor Domain", id=Smsdedbg_img]
+image::Smsdedbg.png[]
+
+The <<MSDCFG>> CSR contains the `sdedbgen` bit that controls whether the
+current scheduled SD is external-debug enabled. This bit is context switched
+(along with rest of the `msdcfg`) per SD. The `medbgen` is an enable control
+for external debug for the M-mode driven by the debug module and is expected
+to be established by the RoT (following RISC-V Security Model recommendation
+SR_GEN_007 and SR_GEN_012). When privilege is `M`, the `medbgen` gates the
+`haltreq_i` from the debug module and if is 0 prevents the hart from entering
+external debug mode. When privilege is less than `M`, the OR of the 
+`MSDCFG.sdedbgen` and `medbgen` gates the `haltreq_i` from the debug module
+and the hart will enter external debug mode if either is 1.
+
+The configuration for `MSDCFG.sdedbgen` is obtained from the manifest of the
+SD and is managed by the M-mode root security manager using secure memory.
+
+When `medbgen` is 1, there are no restrictions. When `medbgen` is 0 but
+`MSDCFG.sdedbgen` is 1, then the external debug mode may be entered but has
+to have restrictions such as `M-mode`-privileged CSR accesses are prohibited
+and Memory accesses by instructions executed in external debug mode cannot
+use `M` privilege. When `medbgen` is 0 and `MSDCFG.sdedbgen` is also 0, then
+the rules around triggers configured with action = 1 that may fire - such as
```

Comment:
In external debug security draft spec, the approach is for Triggers with action = 1. It is not allowed to configure them with action = 1 depending on tdata1.{m,s,u} and debug enables. If they fire as breakpoint exception, it might be a potential security leak (denial of service, privilege escalation attack?).

Reply:
Triggers can be controlled by the RDSM. They will be context switched so this control is only needed for halting external debug.